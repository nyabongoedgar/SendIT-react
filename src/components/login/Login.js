import React from "react";
import { BrowserRouter, Link} from 'react-router-dom';

const Login = (props) => {
  let Loader = require('react-loader')
  return (
    <div className="authForms">
      <form id="login_form" onSubmit={props.submitHandler}>
      <h1>SIGN IN</h1>
        <input
          value={props.username}
          onChange={props.inputHandler}
          type="text"
          name="username"
          placeholder="Username"
          id="login-username"
          required
        />

        <input
          value={props.password}
          onChange={props.inputHandler}
          type="password"
          name="password"
          placeholder="Password"
          id="login-password"
          required
        />
        <Loader loaded={!props.isLoading}>
        <button type="submit" className="btn">
          SUBMIT
        </button>
        </Loader>

        <span>
          <p id="error" />
        </span>

        <p>
          <BrowserRouter>
          Dont have an account ?. 
          <Link to="/signup">Create account</Link> 
          </BrowserRouter>
        </p>
      </form>
    </div>
  );
};

export default Login;
