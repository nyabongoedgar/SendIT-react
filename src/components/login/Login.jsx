import React from "react";

const Login = props => {
  return (
    <div className="container">
      <form id="login_form" onSubmit={props.submitHandler}>
        <input
          value={props.username}
          onChange={props.inputHandler}
          type="text"
          name="username"
          placeholder="Username"
          id="login-username"
          required
        />

        <input
          value={props.password}
          onChange={props.inputHandler}
          type="password"
          name="password"
          placeholder="Password"
          id="login-password"
          required
        />

        <button type="submit" className="btn">
          SUBMIT
        </button>

        <span>
          <p id="error" />
        </span>

        <p>
          Dont have an account ?.
          <a href="register.html">Create account</a>
        </p>
      </form>
    </div>
  );
};

export default Login;
