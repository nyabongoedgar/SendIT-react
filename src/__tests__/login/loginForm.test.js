import React from 'react';
import Login from '../../components/login/Login';
import renderer from 'react-test-renderer';


describe('SignupForm Component', () => {

  it('it matches the snapshot', () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
  });


});
