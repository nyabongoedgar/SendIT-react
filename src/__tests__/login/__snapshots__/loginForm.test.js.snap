// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupForm Component it matches the snapshot 1`] = `
<div
  className="authForms"
>
  <form
    id="login_form"
  >
    <h1>
      SIGN IN
    </h1>
    <input
      id="login-username"
      name="username"
      placeholder="Username"
      required={true}
      type="text"
    />
    <input
      id="login-password"
      name="password"
      placeholder="Password"
      required={true}
      type="password"
    />
    <div
      className="loadedContent"
    >
      <button
        className="btn"
        type="submit"
      >
        SUBMIT
      </button>
    </div>
    <span>
      <p
        id="error"
      />
    </span>
    <p>
      Dont have an account ?.
      <a
        href="/signup"
        onClick={[Function]}
      >
        Create account
      </a>
    </p>
  </form>
</div>
`;
